core/src/main/java/org/jruby/ext/ripper/StringTerm.java

```diff
@@ -146,7 +146,7 @@ public int parseString(RipperLexer lexer, LexerSource src) throws IOException {
     }
 
     private String parseRegexpFlags(RipperLexer lexer, LexerSource src) throws IOException {
-        StringBuilder buf = new StringBuilder(end);
+        StringBuilder buf = new StringBuilder();
```

Comment:
I don't understand this change.

Reply:
I made a general comment in the PR about this.  The result of that string ends up appended to another StringBuilder (which indicates perhaps this could get cleaned up) but 'end' should definitely be in the string.  It was likely me assuming character was an acceptable constructor and because this was ported C code we saved the char as an int and confusion ensued.

Reply:
@nglorioso Actually just remove this.  The existing code should be passing in the StringBuilder from the previous builder in.  I will refactor this a tiny bit and that StringBuilder will go away.

Reply:
Addressed in 58cd1331cb7bdc478b60d29d554f4206fd36e981 on jruby-9.1 and cpd over to master.